src/vehicle.cpp

```diff
@@ -1681,8 +1679,8 @@ bool vehicle::is_connected( const vehicle_part &to, const vehicle_part &from,
         discovered.pop_front();
         auto current = current_part.mount;
 
-        for( int i = 0; i < 4; i++ ) {
-            point next = current + vehicles::cardinal_d[i];
+        for( const point &cardinal_direction : five_cardinal_directions ) {
```

Comment:
The original loop only goes through the first 4 directions. (It stops after `i == 3`, the loop body is not run when `i == 4`). Applies to the loops below as well.

Reply:
Thanks. I missed that - I should use a different point container (w/o `point_zero`) here and below then.